### PR TITLE
Invert if/else to bring precondition early and reduce nesting

### PR DESCRIPTION
**Pull request by drewnoakes**
Following [this comment](https://github.com/gitextensions/gitextensions/pull/4615#discussion_r173673944) from @sharwell, this PR inverts a few if/else statements in order to return early from methods and reduce nesting.

What did I do to test the code and ensure quality:
 - Manual review of automated changes
 - Unit tests
 - Manual testing

Has been tested on:
 - GIT 2.16
 - Windows 10

### CODE REVIEW

File: GitCommands/Git/GitModule.cs [hunk 1]
```diff
@@ -633,14 +633,12 @@ public string RunCacheableCmd(string cmd, string arguments = "", Encoding encodi
...
             {
-                return StripAnsiCodes(EncodingHelper.DecodeString(cmdout, cmderr, ref encoding));
-            }
-
-            GitCommandHelpers.RunCmdByte(cmd, arguments, WorkingDir, null, out cmdout, out cmderr);
+                GitCommandHelpers.RunCmdByte(cmd, arguments, WorkingDir, null, out cmdout, out cmderr);
 
-            GitCommandCache.Add(arguments, cmdout, cmderr);
+                GitCommandCache.Add(arguments, cmdout, cmderr);
+            }
```
Comment (RussKie) [inline GitCommands/Git/GitModule.cs]: to me this looks the inversion of what it said to achieve - the original had less nesting and early return

File: GitUI/Editor/FileViewer.cs [hunk 2]
```diff
@@ -666,29 +666,27 @@ private string GetFileText(string fileName)
...
+                    var content = reader.ReadToEnd();
+                    FilePreamble = reader.CurrentEncoding.GetPreamble();
+                    return content;
                 }
             }
-            else
+            finally
             {
-                return null;
+                fileStream?.Dispose();
```
Comment (RussKie) [inline GitUI/Editor/FileViewer.cs]: a separate piece of work - to replace `try/finally` with `using`

File: GitCommands/Git/GitModule.cs (hunk 1 above)
Comment (drewnoakes) [reply GitCommands/Git/GitModule.cs]: The intended improvement here was the removal of the duplicated `return` line. Note this was part of a secondary commit titled "Invert condition to remove duplicated return expression".

File: GitUI/Editor/FileViewer.cs (hunk 2 above)
Comment (drewnoakes) [reply GitUI/Editor/FileViewer.cs]: Done in a4f62fed98a5d6bd9ae6fc223585f79247b8a66f.

**Comment by drewnoakes** [comment]
@RussKie I think I've addressed your concerns. Let me know if not.

File: GitCommands/Git/GitModule.cs (hunk 1 above)
Comment (RussKie) [reply GitCommands/Git/GitModule.cs]: ahh, right, I see it now.
I looked at the combined diff.

**Pull request merged**